### PR TITLE
Introduce XKit.blog_listener for cacheless blog fetching

### PR DESCRIPTION
**Pull request by AprilSylph**
many users have been complaining about switching accounts needing a refresh for One-Click Postage to get a correct list of blogs again, and as helpfully pointed out by one ask, this has been occurring since XKit Main 2.0.0 was released

turns out, the problem is that xkit main now boots everything too fast! XKit Patches handles the blog list scraping, but XKit Main is now running everything so quickly that One-Click Postage gets booted before the blog scraper code completes its task - but of course once it's done, it stores the result, allowing consecutive `get_blogs` calls to behave properly.

this is fixed by binding OCP's initialisation to the blog message listener, which I've moved to the new `XKit.blog_listener` object to better organise related data

### CODE REVIEW

**Comment by AprilSylph** [comment]
same bug/solution for Mutual Checker; can't see any other extensions using `get_blogs` immediately

**Comment by nightpool** [comment]
delay feels really gross here. can we use promises instead?

File: Extensions/mutualchecker.js [hunk 1]
```diff
@@ -67,7 +71,7 @@ XKit.extensions.mutualchecker = new Object({
 				}
 				XKit.extensions.mutualchecker.check(json_obj, $post.find(".post_info_submissions").first());
 			} else {
-				$post.find(".post_info_link[data-tumblelog-popover]:not(.reblog_icon + .post_info_link), .post-info-tumblelog > a[data-tumblelog-popover]").each(function() {
+				$post.find(".post_info_link[data-tumblelog-popover]:not(.reblog_icon + .post_info_link):not(.tf2_icon + .post_info_link), .post-info-tumblelog > a[data-tumblelog-popover]").each(function() {
```
Comment (nightpool) [inline Extensions/mutualchecker.js]: what's going on with this change?

Comment (AprilSylph) [reply Extensions/mutualchecker.js]: prevents mutualchecker from checking "reblogged from" URLs when TF2 reblogs has already completed on the post

**Comment by AprilSylph** [comment]
rewriting `get_blogs` as a promise function will be... a lot of work but it is a good idea so I'll get right on that

**Comment by AprilSylph** [comment]
...would resolve/reject even work in an event listener callback, actually? it doesn't... seem like it should

**Comment by AprilSylph** [comment]
hmmmmmm i think i know what to do, it's not promises but it IS a much better alternative to delays

**Comment by AprilSylph** [comment]
cleanest method I can think of is giving the blog message listener the ability to run callbacks like the post listener, which isn't quite a Promise

But It's Not A Delay:tm:

**Pull request merged**